openpype/tools/sceneinventory/view.py

```diff
@@ -146,25 +142,25 @@ def _build_item_menu_for_selection(self, items, menu):
         switch_to_versioned = None
         if has_loaded_hero_versions:
             def _on_switch_to_versioned(items):
-                repre_ids = []
+                repre_ids = set()
                 for item in items:
-                    item_id = ObjectId(item["representation"])
+                    item_id = item["representation"]
                     if item_id not in repre_ids:
-                        repre_ids.append(item_id)
+                        repre_ids.add(item_id)
 
                 repre_docs = get_representations(
                     project_name,
                     representation_ids=repre_ids,
                     fields=["parent"]
                 )
 
-                version_ids = []
+                version_ids = set()
```

Comment:
It's not a newly introduced bug but it does look like confusing code. On line 171 `version_ids` is redefined as a `set` and from a quick glance over the code I'm not entirely sure why it needs a completely new clear definition like that?

Should the attribute on 171 be called `hero_version_ids`, etc.?
Why does it need to be rebuild like that there from hero versions? A comment might be welcome?

Reply:
Changed to `hero_src_version_ids`.